src/test/java/io/specto/hoverfly/junit/core/HoverflyTest.java

```diff
@@ -711,7 +719,9 @@ public void tearDown() {
 
     private HoverflyClient createMockHoverflyClient(Hoverfly hoverfly) {
         HoverflyClient hoverflyClient = mock(HoverflyClient.class);
-        Whitebox.setInternalState(hoverfly, "hoverflyClient", hoverflyClient);
+        HoverflyClientFactory clientFactory = mock(HoverflyClientFactory.class);
+        when(clientFactory.createHoverflyClient(any(HoverflyConfiguration.class))).thenReturn(hoverflyClient);
+        Whitebox.setInternalState(hoverfly, "hoverflyClientFactory", clientFactory);
```

Comment:
Mocking hoverflyClientFactory seems redundant because you are not verifying its behaviours, and complicating some unit tests which now requires setting up:
```
when(hoverflyClient.getHealth()).thenReturn(true);
hoverfly.start();
```